packages/aws-cdk/lib/notices.ts

```diff
@@ -485,13 +488,18 @@ export interface NoticeDataSource {
 export class WebsiteNoticeDataSource implements NoticeDataSource {
   private readonly options: SdkHttpOptions;
 
-  constructor(private readonly ioMessages: IoDefaultMessages, options: SdkHttpOptions = {}) {
+  constructor(private readonly ioHelper: IoHelper, options: SdkHttpOptions = {}) {
     this.options = options;
   }
 
-  fetch(): Promise<Notice[]> {
+  async fetch(): Promise<Notice[]> {
     const timeout = 3000;
-    return new Promise((resolve, reject) => {
+
+    const options: RequestOptions = {
+      agent: await new ProxyAgentProvider(this.ioHelper).create(this.options),
+    };
+
+    const notices = await new Promise<Notice[]>((resolve, reject) => {
```

Comment:
bit of a rejig here to allow sending a message after the promise resolved

Reply:
Could just have done a `.then()` at the end 🤣

Reply:
Happy to change it, I don't mind either way. I've received feedback in the past that mixing promise styles is confusing.